src/qibo/tomography/gate_set_tomography.py

```diff
@@ -97,8 +96,8 @@ def _get_observable(j: int, nqubits: int):
     return SymbolicHamiltonian(observable, nqubits=nqubits)
 
 
-@cache
-def _prepare_state(k, nqubits):
+# @cache
+def _prepare_state(k: int, nqubits: int):
```

Comment:
```suggestion
def _prepare_state(index: int, nqubits: int):
```

We should avoid non-descriptive names to variables.

Reply:
I think it's ok to use `k` for `_prepare_state` and `j` for `_measurement_basis` because the docstring describes specifically the ranges for `k` and `j`, depending on how `nqubits`.